include/swift/Basic/Dwarf.h

```diff
@@ -21,7 +21,7 @@
 
 namespace swift {
   /// The DWARF version emitted by the Swift compiler.
-  const unsigned DWARFVersion = 3;
+  const unsigned DWARFVersion = 4;
```

Comment:
My dev environment was cygwin 64bit 2.3.1, and clang version was 3.5.2.
The dwarf version 4 is used as default and the linker displayed errors when
compiled with dwarf version 3.

-----
$ clang++ hello.cpp -gdwarf-3
/tmp/hello-c8fc34.o:(.debug_info+0x16): relocation truncated to fit:
R_X86_64_32 against `.debug_line'
collect2: error: ld returned 1 exit status
clang: error: linker (via gcc) command failed with exit code 1 (use -v to
see invocation)
-----

So swiftc -c -g must generate in dwarf ver 4.

Is this should be changed to like this?

-----
#if __CYGWIN__
  const unsigned DWARFVersion = 4;
#else
  const unsigned DWARFVersion = 3;
#fi
-----


2016-01-19 9:10 GMT+09:00 Dmitri Gribenko <notifications@github.com>:

> In include/swift/Basic/Dwarf.h
> <https://github.com/apple/swift/pull/1010#discussion_r50059136>:
>
> > @@ -21,7 +21,7 @@
> >
> >  namespace swift {
> >    /// The DWARF version emitted by the Swift compiler.
> > -  const unsigned DWARFVersion = 3;
> > +  const unsigned DWARFVersion = 4;
>
> Why was this required? /cc @adrian-prantl
> <https://github.com/adrian-prantl>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apple/swift/pull/1010/files#r50059136>.
>